src/snowcli/snowsql_config.py

```diff
@@ -28,3 +29,12 @@ def add_connection(self, connection_name, entry):
 
         with open(os.path.expanduser(self.path), "w+") as f:
             self.config.write(f)
+
+    def _replace_with_env_variables(self):
+        connection_sections = [
+            s for s in self.config.sections() if s.startswith("connections.")
+        ]
+        for section in connection_sections:
+            for (name, value) in self.config[section].items():
+                if value.startswith("$SNOWCLI_"):
+                    self.config[section][name] = os.environ[value[1:]]
```

Comment:
Hm, I'm wondering if wouldn't it be better to resolve the variables when we read them?

Reply:
You mean just before pass them to snowflake connector?

Reply:
1. Why do we have to limit env substitution only to `connections` sections? IMO the code will be simpler if we apply env variables resolution to whole config (even if we don't need the rest of the config).
2. Do I understand the code correctly that it won't be possible to use env variable in the middle of string value (that value has to start with "$SNOWCLI_"?
3. Maybe we can somehow add env vars resolution to configparser (I found a few ideas here: https://stackoverflow.com/questions/26586801/configparser-and-string-interpolation-with-env-variable)?  It might be more consistent with the syntax of other interpolations already possible to use in the config?

Reply:
ad1. Because we only need and use connections part
ad2. Yes, it has start with `$SNOWCLI_`. But why someone would like to use it in the middle of string? 
ad3. Configparser with os.environ loads all env variables into default section in config object. I don't like it :/

Reply:
1. Ok, so if we use only connections part, then maybe the config should not be named `SnowsqlConfig`? I understand naming of this class can be outside of the scope of this PR, but then maybe we can at least filter out all unused sections just after creation of config object instead of doing it in the function responsible for env vars resulution (which imo shouldn't have to care which sections are used and which not).
2. If we care only about currently used settings, then I can't give any example but I can imagine using it in paths, urls, ... So, if the goal is to handle env vars only for current form of snowsql connection configuration, then the solution is ok, but if the goal is to prepare mechanism for the future usage, then IMO we're limiting this usage.
3. Two notes here:
    * `ConfigParser(os.environ)` is only one of solutions.
    * We can filter env vars dictionary searching for some prefix.

Reply:
1. Changed `SnowsqlConfig` to `ConnectionConfigs`
2. We can extend this in future, for now it is not needed
3. Checked this solution a bit more and configparser coping default variables into all sections. It creates only mess in dict

Reply:
I think we should start simple, applying this only to connection sections is fine. Later on we can consider extending this to more than env vriables - for example reading values from external vaults.

Here are something that can inspire us https://github.com/apache/airflow/blob/bc1d68a6eb01919415c399d678f491e013eb9238/airflow/configuration.py#L622-L667

> You mean just before pass them to snowflake connector
Yes. Basically every time we access a config variable we should evaluate if it's not present in environment.

Reply:
@sfc-gh-astus 
1. Ok. I left a few other separate comments related to this rename.
2. Ok.
3. Ok.